### PR TITLE
Filebeat scrape specific annotation only

### PR DESCRIPTION
**Pull request by sincejune**
filebeat in this package will scrape pod with annotation `dashbase.io/scrape="true"`.
 
```
filebeat:
  elasticsearch_url: https://proxy:9200
```
with this section in top level, we will enable this.

### CODE REVIEW

**Comment by keitaf** [comment]
@khou and I talked about whether we should include filebeat in our Dashbase chart, and we both agreed that filebeat should be a separate chart. So can you please make this change into a independent chart, and make it configurable such that you can change ES endpoint and annotation(s) to choose which Pods to harvest logs from?

Ideally we want to support the following use cases:
1) deploy a filebeat chart once to collect logs from all Pods in the whole cluster to `filebeat` table
2) deploy a filebeat chart to collect logs from specific Pods (e.g., Pods with annotation `app=dashbase, namespace=dashbase`), add extra annotations (e.g., `service=ABC`), and send them to `xyz` table

**Comment by khou** [comment]
I second Keita's thought on splitting Filebeat into a separate chart. Though it feels better as a product, people may want to customize filebeat settings, etc. that I don't feel belongs in our values.yml

**Comment by SinceJune** [comment]
How do you think about the scene I mentioned. @khou @keitaf

**Comment by khou** [comment]
For the use case you mentioned, let's do it! This feels like use for when user isn't using k8s product but Dashbase is deployed on K8s, whereas we'll have another filebeat helm template or what not for K8s logs product

File: dashbase/templates/filebeat.yaml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: filebeat
+  labels:
+    app: {{.Release.Name}}
+    component: filebeat
+spec:
+  selector:
```
Comment (keitaf) [inline dashbase/templates/filebeat.yaml]: can we use `podAffinity` to only deploy this filebeat onto the same node(s) where other Dashbase Pods are running? If the k8s cluster has 1K nodes but Dashbase is only using 5 of them, it's wasteful to deploy this filebeat to all the 1K nodes.

**Comment by SinceJune** [comment]
added podConfig in top level:
```
filebeat:
  enabled: true
  elasticsearch_url: http://proxy:9200
  podConfig:
    nodeSelector:
      cloud.google.com/gke-nodepool: dashbase
```

File: dashbase/templates/filebeat.yaml (hunk 1 above)
Comment (SinceJune) [reply dashbase/templates/filebeat.yaml]: currently, kubernetes daemonset doesn't support podAffinity/podAntiAffinity yet.
we only can use `nodeSelector`/`nodeAffinity` to deploy daemonset.
see https://github.com/kubernetes/kubernetes/issues/29276#issuecomment-418878667

**Comment by sincejune** [comment]
Add Documentation here:
#### Enable filebeat as daemonsets in Dashbase cluster

We now can enable filebeat as a daemon sets inside Dashbase Kubernets cluster. The feature will be disabled by default.

Here's some example about how to enable it:

##### filebeat on all kubernetes nodes

```yaml
dashbase_version: 0.10.9

username: ...
license: ...

https: true

filebeat:
  enabled: true
  elasticsearch_url: https://proxy.example.dashbase.io:9200
```

All `filebeat` section which has specified `elasticsearch_url` will launch filebeat instance onto all kubernetes nodes.

If you do not add `enabled: true` or `elasticsearch_url`, It won't be enabled.



##### filebeat on nodes with specified labels.

```yaml
dashbase_version: 0.10.9

username: ...
license: ...

https: true

filebeat:
  enabled: true
  elasticsearch_url: https://proxy.example.dashbase.io:9200
  podConfig:
    nodeSelector:
      cloud.google.com/gke-nodepool: filebeat
```

For example, you can add [nodeSelector](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector) into filebeat pods using the above config. Also, you can use a more accurate description here, e.g. [affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).

**Pull request merged**